src/app/components/cases/CreateForm/CreateForm.tsx

```diff
@@ -1,35 +1,44 @@
 import { useState, useEffect } from "react"
 import scaffold from "./scaffold"
-import { useCaseThemes, useReasons, useCaseCreate, useProjects } from "app/state/rest"
+import { useCaseThemes, useReasons, useCaseCreate, useProjects, useListing } from "app/state/rest"
 import ConfirmScaffoldForm from "app/components/shared/ConfirmScaffoldForm/ConfirmScaffoldForm"
 import useNavigateWithFlashMessage from "app/state/flashMessages/useNavigateWithFlashMessage"
 import useScaffoldedFields from "app/components/shared/ConfirmScaffoldForm/hooks/useScaffoldedFields"
 
 type Props = {
   bagId: Components.Schemas.Address["bag_id"]
+  tonId: number | undefined
 }
 
 type FormData =
   Pick<CaseCreate, "address" | "description"> &
   { theme: Components.Schemas.CaseTheme, reason: Components.Schemas.CaseReason, project: Components.Schemas.CaseProject }
 
-const mapData = (bagId: Components.Schemas.Address["bag_id"]) =>
+const mapData = (bagId: Components.Schemas.Address["bag_id"], tonId: number | undefined) =>
   (data: FormData): CaseCreate => ({
     ...data,
     address: { bag_id: bagId },
     theme: data.theme.id,
     reason: data.reason.id,
-    project: data.project?.id
+    project: data.project?.id,
+    ton_ids: tonId ? [ tonId ] : undefined
```

Comment:
Would add a explicit check for undefined. `tonId !== undefined ? [ tonId ] : undefined`. Otherwise a zero number value could be posted.